install_test/as_subdirectory/as_subdirectory.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+args=$#
+WK_DIR=/tmp/kokkos_fft_as_subdirectory
+SRC_DIR=/work
+ROOT_DIR=$1
+BUILD_TYPE=$2
+C_COMPILER=$3
+CXX_COMPILER=$4
+
+if [ $args -eq 5 ]; then
+    # CPU build
+    # e.g
+    # /tmp Release gcc g++ -DKokkos_ENABLE_THREADS=ON
+    # TARGET_FLAG is empty
+    BACKEND_FLAG=$5
+    TARGET_FLAG=$6
+elif [ $args -eq 6 ]; then
+    # GPU build without target flag
+    # e.g.
+    # /tmp Release gcc g++ \
+    # -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_AMPERE80=ON 
+    # TARGET_FLAG is empty
+    BACKEND_FLAG0=$5
+    BACKEND_FLAG1=$6
+    BACKEND_FLAG="${BACKEND_FLAG0} ${BACKEND_FLAG1}"
+    TARGET_FLAG=$7
+elif [ $args -eq 7 ]; then
+    # GPU build with target flag
+    # e.g.
+    # /tmp Release gcc g++ \
+    # -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_AMPERE80=ON -DKokkosFFT_ENABLE_HOST_AND_DEVICE=ON
+    BACKEND_FLAG0=$5
+    BACKEND_FLAG1=$6
+    BACKEND_FLAG="${BACKEND_FLAG0} ${BACKEND_FLAG1}"
+    TARGET_FLAG=$7
+else
+    echo "*** Error: Number of arguments must be 5, 6 or 7 ***"
+    exit 1;
+fi
+
+# Make work directory (Project directory)
+mkdir -p ${WK_DIR} && cd ${WK_DIR}
+
+# Copy CMakeLists.txt and hello.cpp
+cp ${SRC_DIR}/install_test/as_subdirectory/* .
+
+# Prepare Kokkos and KokkosFFT under tpls
+mkdir tpls && cd tpls
+
+# Prepare tpls/kokkos 
+git clone https://github.com/kokkos/kokkos.git
+
+# Copy KokkosFFT under tpls/kokkos-fft
+cp -r ${SRC_DIR} kokkos-fft
```

Comment:
Looks very complicated. Why not just go to this directory and have a CMakeLists.txt that looks like https://github.com/kokkos/kokkos/blob/e0dc0128e04f18c2bbbaefceef3616e7ddcfa3c4/example/build_cmake_in_tree/CMakeLists.txt#L10

Reply:
I can try, but not sure it works. Using Kokkos added as a submodule causes an error in the git check. https://github.com/CExA-project/kokkos-fft/actions/runs/8008651458/job/21875544621

In addition, I would like a CMakeLists.txt and build commands exactly the same as used in the docs.

Reply:
This file was finally removed.

Reply:
Great!